addons/base_import_module/models/ir_module.py

```diff
@@ -402,18 +405,40 @@ def button_immediate_install_app(self):
 
     @api.model
     def _get_missing_dependencies(self, zip_data):
-        modules = self._get_missing_dependencies_modules(zip_data)
+        modules, unavailable_modules = self._get_missing_dependencies_modules(zip_data)
         description = ''
-        if modules:
-            description = _('The following modules will be also installed:\n')
+        if unavailable_modules:
+            description = _(
+                "The installation of the data module would fail as the following dependencies can't"
+                " be found in the addons-path:\n"
+            )
+            for module in unavailable_modules:
+                description += "- " + module + "\n"
+            theme_not_found = any(module[:6] == "theme_" for module in unavailable_modules)
+            enterprise_not_found = any(
+                module[:6] != "theme_" and not self.search([('name', '=', module)], limit=1)
```

Comment:
tu peux virer le search

Reply:
or if you really want: not self.search([('name', '=', module), ('to_buy', '=', False)]